src/common/com/intellij/plugins/haxe/ide/annotator/HaxeSemanticAnnotator.java

```diff
@@ -69,10 +69,58 @@ private static void analyzeSingle(final PsiElement element, AnnotationHolder hol
       LocalVarChecker.check((HaxeLocalVarDeclaration)element, holder);
     } else if (element instanceof HaxeStringLiteralExpression) {
       StringChecker.check((HaxeStringLiteralExpression)element, holder);
+    } else if (element instanceof HaxeTypeCheckExpr) {
+      TypeCheckExpressionChecker.check((HaxeTypeCheckExpr)element, holder);
+    }
+
+  }
+}
+
+class TypeCheckExpressionChecker {
+  public static void check(
+    final HaxeTypeCheckExpr expr,
+    final AnnotationHolder holder
+  ) {
+    final PsiElement[] children = expr.getChildren();
+    if (children.length == 2) {
+      final HaxeGenericResolver resolver = HaxeGenericResolverUtil.generateResolverFromScopeParents(expr);
+      final ResultHolder statementResult = HaxeTypeResolver.getPsiElementType(children[0], expr, resolver);
+      ResultHolder assertionResult = SpecificTypeReference.getUnknown(expr).createHolder();
+      if (children[1] instanceof HaxeTypeOrAnonymous) {
+        assertionResult = HaxeTypeResolver.getTypeFromTypeOrAnonymous((HaxeTypeOrAnonymous)children[1]);
+        ResultHolder resolveResult = resolver.resolve(assertionResult.getType().toStringWithoutConstant());
+        if (null != resolveResult) {
+          assertionResult = resolveResult;
+        }
+      }
+      if (!assertionResult.canAssign(statementResult)) {
+        final HaxeDocumentModel document = HaxeDocumentModel.fromElement(expr);
+        Annotation annotation = holder.createErrorAnnotation(children[0],
```

Comment:
It's better move that string to resources.

Reply:
Done.